### PR TITLE
Threshold: Add matrix of state change behavior

### PR DESCRIPTION
**Pull request by Gitsarry**
Since it was not clear to me from reading the threshold documentation for what upper/lower limits with/without hysteresis exactly a state change would occur, here is the actual behavioral matrix (excluding negative hysteresis, not sure what that should be or if it is really even supported), which I tested with HA 2024.7.2.

Matrix tested with value of hysteresis >= 0

See https://community.home-assistant.io/t/threshold-integration-matrix-of-state-change-behavior/756257

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Introduced a "Matrix of state change behavior" section in the documentation, detailing sensor state changes based on readings.
	- Added tables for "Sensor value rising" and "Sensor value falling," clarifying conditions for sensor activation and deactivation.

- **Documentation**
	- Enhanced clarity and usability of the documentation regarding sensor operational logic.

### CODE REVIEW

File: source/_integrations/threshold.markdown [hunk 1]
```diff
@@ -76,6 +76,26 @@ name:
...
+## Matrix of state change behavior
+
+### Sensor value rising
+```
+set             turns on when             turns off when
+---             -------------             --------------
+only upper      sensor > (upper + hyst)   never
+only lower      never                     sensor > (lower + hyst)
+upper & lower   sensor > (lower + hyst)   sensor > (upper + hyst)
+```
```
Comment (frenck) [inline source/_integrations/threshold.markdown]: Let's make use of tables here, instead of trying to add tables in a code fenced block.

**Review by frenck — CHANGES_REQUESTED** [review]

**Comment by Gitsarry** [comment]
Hope it is good now. How could I test how it looks in my clone repository?

**Comment by Gitsarry** [comment]
> Hope it is good now. How could I test how it looks in my clone repository?

Looks quite ok on https://markdownlivepreview.com/

File: source/_integrations/threshold.markdown [hunk 2]
```diff
@@ -76,6 +76,24 @@ name:
   default: Threshold
 {% endconfiguration %}
 
+## Matrix of state change behavior
+
+### Sensor value rising
+
+| set           | turns on when           | turns off when          |
```
Comment (frenck) [inline source/_integrations/threshold.markdown]:
```suggestion
| Set           | Turns on when           | Turns off when          |
```

File: source/_integrations/threshold.markdown [hunk 3]
```diff
@@ -76,6 +76,24 @@ name:
...
+| ------------- | ----------------------- | ----------------------- |
+| only upper    | sensor > (upper + hyst) | never                   |
+| only lower    | never                   | sensor > (lower + hyst) |
+| upper & lower | sensor > (lower + hyst) | sensor > (upper + hyst) |
+
+### Sensor value falling
+
+| set           | turns on when           | turns off when          |
```
Comment (frenck) [inline source/_integrations/threshold.markdown]:
```suggestion
| Set           | Turns on when           | Turns off when          |
```

**Review by frenck — APPROVED** [review]
Thanks, @Gitsarry 👍 

../Frenck

**Pull request merged**